packages/office-ui-fabric-react/src/components/DetailsList/DetailsRow.tsx

```diff
@@ -200,6 +203,30 @@ export class DetailsRow extends BaseComponent<IDetailsRowProps, IDetailsRowState
     const showCheckbox = selectionMode !== SelectionMode.none && checkboxVisibility !== CheckboxVisibility.hidden;
     const ariaSelected = (selectionMode === SelectionMode.none) ? undefined : isSelected;
 
+    const rowFields = (
+      <RowFields
+        columns={ columns }
+        item={ item }
+        itemIndex={ itemIndex }
+        columnStartIndex={ showCheckbox ? 1 : 0 }
+        onRenderItemColumn={ onRenderItemColumn }
+        shimmer={ shimmer }
+      />
+    );
+    // Rendering Shimmer Animation outside the focus zone
+    if (shimmer) {
```

Comment:
Is the row component the right place for this? Or should we go one level higher and just not render the row and instead render that shimmer component from the detailslist?

Reply:
We intend to use DetailsRow in the host application so that we can leverage the DetailsRowFields styling to look as we have columns with shimmer lines not just a single line for the entire width of the row. So we need this in here. Eventually I think the shimmer will be moved inside the detailsList so that we are not exposing DetailsRow to the host.

Reply:
@cschleiden As of now the Shimmer component can not be imported to DetailsList because it's part of experiments package, which is not a dependency of office-ui-fabric-react.
When the shimmer will be mature enough to be moved along side with other Fabric components I intend to use it internally in DetailsList and even then we should check if the bundle size of the DetailsList is going to be something to worry about.

Reply:
Wait, but shimmer will still be included in the bundle?

Reply:
You're right my bad. I guess then I don't really understand your approach for including it in detailslist... Why do we have to have that 10 item array there if the rendering is the user's responsibility? I could just do 

```
items={isLoading ? Array(10).fill(null) : realItems}
```

wherever I use the List? And nothing would have to change

Reply:
@cschleiden 
I think we took this decision for consistency. User should not decide how many lines of shimmer should be rendered in the beginning. DetailsList will always give them 10 lines as we don't know how many realItems will eventually be rendered. I also will add a fading-out to bottom applying a decreasing opacity in styles based on this number (this will remove the expectancy of the user of how many items to wait even if we have 10 lines of shimmer). Without mergeStyle I can not pass a number the user will give me to .sass, so if I know a hardcoded number I could do a loop inside .sass and apply this opacity.

P.S. this number could change as soon as we can demo this to designers.